Simplenote/SPAppDelegate+Extensions.swift

```diff
@@ -405,3 +406,61 @@ extension SPAppDelegate {
         EditorFactory.shared.scrollPositionCache.cleanup(keeping: allIdentifiers)
     }
 }
+
+// MARK: - Background Fetch
+//
+@available(iOS 13.0, *)
+extension SPAppDelegate {
+    @objc
+    func registerBackgroundRefreshTask() {
+        guard BuildConfiguration.current == .debug else {
+            return
+        }
+
+        NSLog("Registered background task with identifier \(BackgroundRefreshConstants.bgTaskIdentifier)")
+        BGTaskScheduler.shared.register(forTaskWithIdentifier: BackgroundRefreshConstants.bgTaskIdentifier, using: .main) { task in
+            guard let task = task as? BGAppRefreshTask else {
+                return
+            }
+            self.handleAppRefresh(task: task)
+        }
+    }
+
+    private func handleAppRefresh(task: BGAppRefreshTask) {
+        NSLog("Did fire handle app refresh")
+        guard BuildConfiguration.current == .debug else {
+            return
+        }
+
+        NSLog("Background refresh intiated")
+        scheduleAppRefresh()
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + BackgroundRefreshConstants.timeOut) {
+            NSLog("Background refresh finishing")
+            task.setTaskCompleted(success: true)
+        }
+    }
```

Comment:
We should also set an expiration handler, because according to [Apple Docs](https://developer.apple.com/documentation/backgroundtasks/bgtask/3142236-settaskcompleted):
> If you don’t set an expiration handler, the system will mark your task as complete and unsuccessful instead of sending a warning.

If we set an expiration handler I think there is no reason to have our own timeout.

---
Other ideas:
* We can extract all BG related code to a separate class to clean up app delegate
* To avoid always waiting for BG task to expire we can try to be smart: re-set a timer for x seconds on every change that we receive (`- (void)bucket:(SPBucket *)bucket didChangeObjectForKey:(NSString *)key....`). So when all changes are received and processed, we wait a bit and finish BG task.